src/main/java/com/gamingmesh/jobs/PlayerManager.java

```diff
@@ -881,32 +881,43 @@ public BoostMultiplier getInventoryBoost(Player player, Job prog) {
 	if (player == null || prog == null)
 	    return data;
 
-	ItemStack iih = Jobs.getNms().getItemInMainHand(player);
-	JobItems jitem = getJobsItemByNbt(iih);
-	if (jitem != null && jitem.getJobs().contains(prog))
-	    data.add(jitem.getBoost(getJobsPlayer(player).getJobProgression(prog)));
-
-	// Lets check offhand
-	if (Version.isCurrentEqualOrHigher(Version.v1_9_R1) && Jobs.getGCManager().boostedItemsInOffHand) {
-	    iih = CMIReflections.getItemInOffHand(player);
-	    if (iih != null) {
-		jitem = getJobsItemByNbt(iih);
-		if (jitem != null && jitem.getJobs().contains(prog))
-		    data.add(jitem.getBoost(getJobsPlayer(player).getJobProgression(prog)));
-	    }
-	}
+	ItemStack iih;
+	List<JobItems> jitems = new ArrayList<>();
 
-	for (ItemStack oneArmor : player.getInventory().getArmorContents()) {
-	    if (oneArmor == null || oneArmor.getType() == org.bukkit.Material.AIR)
-		continue;
+    // Check mainhand slot
+    if (Jobs.getGCManager().boostedItemsInMainHand) {
+        iih = Jobs.getNms().getItemInMainHand(player);
 
-	    JobItems armorboost = getJobsItemByNbt(oneArmor);
-	    if (armorboost == null || !armorboost.getJobs().contains(prog))
-		continue;
+        if (iih != null) {
+            jitems.add(getJobsItemByNbt(iih));
+        }
+    }
+
+    // Check offhand slot
+    if (Version.isCurrentEqualOrHigher(Version.v1_9_R1) && Jobs.getGCManager().boostedItemsInOffHand) {
+        iih = CMIReflections.getItemInOffHand(player);
 
-	    data.add(armorboost.getBoost(getJobsPlayer(player).getJobProgression(prog)));
+        if (iih != null) {
+            jitems.add(getJobsItemByNbt(iih));
+        }
+    }
+
+    // Check armor slots
+	if (Jobs.getGCManager().boostedArmorItems) {
+		for (ItemStack oneArmor : player.getInventory().getArmorContents()) {
+			if (oneArmor == null || oneArmor.getType() == org.bukkit.Material.AIR)
+				continue;
+
+			jitems.add(getJobsItemByNbt(oneArmor));

```

Comment:
Once we are here, reduce the complexity of this for loop.
```suggestion
			if (oneArmor != null && oneArmor.getType() != org.bukkit.Material.AIR) {
			    jitems.add(getJobsItemByNbt(oneArmor));
			}
```